staking/cli/src/instructions.rs

```diff
@@ -279,19 +297,64 @@ pub fn process_write_encoded_vaa(
     process_transaction(
         rpc_client,
         &[
-            create_encoded_vaa,
-            init_encoded_vaa_instruction,
-            write_encoded_vaa_accounts_instruction,
             verify_encoded_vaa_instruction,
             request_compute_units_instruction,
         ],
-        &[payer, &encoded_vaa_keypair],
+        &[payer],
     );
 
 
     encoded_vaa_keypair.pubkey()
 }
 
+pub fn write_encoded_vaa(
+    rpc_client: &RpcClient,
+    payer: &Keypair,
+    encoded_vaa: &Pubkey,
+    wormhole: &Pubkey,
+    chunk: &[u8],
+) {
+    let write_encoded_vaa_accounts = wormhole_core_bridge_solana::accounts::WriteEncodedVaa {
+        write_authority: payer.pubkey(),
+        draft_vaa:       *encoded_vaa,
+    }
+    .to_account_metas(None);
+
+    let write_encoded_vaa_accounts_instruction = Instruction {
+        program_id: *wormhole,
+        accounts:   write_encoded_vaa_accounts.clone(),
+        data:       wormhole_core_bridge_solana::instruction::WriteEncodedVaa {
+            args: WriteEncodedVaaArgs {
+                index: 0,
```

Comment:
should this index be more than zero after the first instruction?

Reply:
mmmm yes!

Reply:
fixed